app/controllers/pieces_controller.rb

```diff
@@ -4,22 +4,24 @@ def update
     @piece = Piece.find(params[:id])
     @game = Game.find(@piece.game_id)
 
-    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i)
+    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i) && !@piece.moving_into_check?(params[:y_coord].to_i, params[:x_coord].to_i) 
```

Comment:
I see that if the game is already in check before we call moving_into_check? we will get that we are moving into check regardless of whether it has anything to do with the piece we are moving. I am trying to think of the best way to fix that...